src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,278 @@
+        }
+
+        return GetMachineTranslationIdAndText(machineTranslations, machineTranslationId);
+    }
+    
+    private async Task<(int, string)> GetMachineTranslationTextAndIdForSnapshot(
+        int resourceContentVersionSnapshotId,
+        int? machineTranslationId,
+        CancellationToken ct)
+    {
+        var machineTranslations = await _dbContext
+                                            .ResourceContentVersionMachineTranslations
+                                            .Where(x => x.ResourceContentVersion.ResourceContentVersionSnapshots
+                                                .Any(s => s.Id == resourceContentVersionSnapshotId))
+                                            .OrderBy(x => x.Id)  
+                                            .ToListAsync(ct);
+
+        if (machineTranslations.Count == 0)
+        {
+            throw new InvalidOperationException(
+                $"No Machine translation for Snapshot with id {resourceContentVersionSnapshotId} found");
+        }
+
+        if (machineTranslationId is not null && !machineTranslations.Any(m => m.Id != machineTranslationId))
+        {
+            throw new InvalidOperationException(
+                $"Machine translation with id {machineTranslationId} not found for snapshot version {resourceContentVersionSnapshotId}");
+        }
+        
+        return GetMachineTranslationIdAndText(machineTranslations, machineTranslationId);
+    }
+
+    private (int, string) GetMachineTranslationIdAndText(
+        List<ResourceContentVersionMachineTranslationEntity> machineTranslations,
+        int? machineTranslationId)
+    {
+        List<string> contentList;
+        string machineTranslationText;
+        
+        if (machineTranslations.Count == 1)
+        {
+            return (
+                machineTranslations.First().Id, 
+                HtmlUtilities.GetPlainText(machineTranslations.First().Content));
+        }
+
+        // We want to ensure that we get the correct set of MTs, whether it is 1 or many ( as in the case of FIA content )
+        // So, we determine the reference MT. This can be the specified MT or the latest MT with a ContentIndex of 0.
+        var referenceMachineTranslation = machineTranslations
+            .Where(x => x.ContentIndex == 0 && (machineTranslationId is null || x.Id == machineTranslationId))
+            .OrderByDescending(x => x.Id)
+            .FirstOrDefault()
+            ?? throw new InvalidOperationException("No Machine translation with content index 0 found");
+
+        if (machineTranslationId is null)
+        {
+            // Assume we are using the latest MT
+            contentList = machineTranslations
+                .Where(x => x.Id >= referenceMachineTranslation.Id)
+                .Select(x => x.Content)
+                .ToList();
+            
+            machineTranslationText = HtmlUtilities.GetPlainText(string.Join(string.Empty, contentList));
```

Comment:
It's possible that `contentList` could be an empty list here so I suggest throwing an exception instead of operating on an empty string.